arch/src/x86_64/mod.rs

```diff
@@ -237,6 +241,264 @@ pub enum CpuidReg {
+                    let idx = missing_bits.trailing_zeros() as usize;
+                    bit_positions.push(u8::try_from(idx).expect(
+                        "idx is at most 32 hence it can be represented as a u8 without problems",
+                    ));
+                    let least_significant_bit = missing_bits & missing_bits.wrapping_neg();
+                    missing_bits ^= least_significant_bit;
+                }
+                // TODO: Use a proper register name rather than REG and consider returning this as an error instead of logging here
+                log::warn!(
+                    "the given cpuid entry identified by: \n
+                 function = 0x{:08x} \
+                 index = 0x{:08x} \
+                 flags =  0x{:08x} \
+                 does not have the following bits set: {:?} in the register {:?}
+                 even though the specified restriction permits it.
+                ",
+                    entry.function,
+                    entry.index,
+                    entry.flags,
+                    bit_positions,
+                    REG
+                );
+            }
+        } else {
+            // TODO: Better log or return an error here
+            log::warn!("no entry matched");
+        }
+    }
+}
+
+/// Reduces boilerplate when implementing CpuIdFeatureFlag constants.
+/// One passes in a parameterized CpuIdFeatureFlags together with
+/// a name per bit in the 32bit bitset.
+///
+/// For lower order bits that should not have any associated constant, appearing
+/// before a higher order bit that needs a constant, you can simply place a "NULL"
+/// in its place.
+macro_rules! cpuid_flag_constants {
+    /* NOTE: We allow dead code within this macro as we may want to use certain
+     unused constants in the feature when introducing more cpu profiles.
+     The alternative would be to fill the macro invocations with more "NULL"
+     entries.
+    */
+    //============ Possible starting points ===========//
+    ($t:ty, $name:ident) => {
+        impl $t {
+            #[allow(dead_code)]
+            const $name: Self = Self(1);
+        }
+    };
+    ($t:ty, "NULL", $($tail:tt)*) => {
+        impl $t {
+            cpuid_flag_constants!(1, $($tail)*);
+        }
+    };
+    ($t:ty, $name:ident, $($tail:tt)*) => {
+        impl $t {
+            #[allow(dead_code)]
+            const $name: Self = Self(1);
+            cpuid_flag_constants!(1, $($tail)*);
+        }
+    };
+
+    //============ Possible most deeply nested macro invocation ===========//
+    ($i:expr, $name:ident) => {
+        #[allow(dead_code)]
+        const $name: Self = Self(1 << $i);
+    };
+
+    ($i:expr, "NULL") => {};
+
+    ("NULL") => {};
+
+    () => {};
+
+    // Also permit ending with a trailing ,
+    ($i:expr, $name:ident,) => {
+        #[allow(dead_code)]
+        const $name: Self = Self(1 << $i);
+    };
+
+    ($i:expr, "NULL",) => {};
+
+    ("NULL",) => {};
+
+    (,) => {};
```

Comment:
TODO: The trailing comma can be made optional in the same matchers as above, hence this may be removed.